mako/utils/transformers.cc

```diff
@@ -0,0 +1,106 @@
+// Adapted from https://github.com/vllm-project/vllm/blob/v0.2.7/vllm/model_executor/weight_utils.py
+// Copyright 2024 The Mako Authors
+// Copyright 2023 The vLLM team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mako/utils/transformers.h"
+
+#include <algorithm>
+
+/// \brief Utility to find weight files from model directory
+/// \param model_name_or_path A path to a directory containing model weights saved using
+/// \param cache_dir Path to the folder where cached files are stored.
+/// \param load_format Format of the model to load. Must be one of ``"auto"``, ``"safetensors"``, ``"pt"``, or ``"npcache"``.
+/// \param fall_back_to_pt If ``true``, will always allow pt format.
+/// \param revision An optional Git revision id which can be a branch name, a tag, or a commit hash.
+/// \return Pairs of name and weight of the loaded model.
+static std::tuple<std::string, std::vector<std::string>, bool> prepare_hf_model_weights(
+        std::string model_name_or_path,
+	std::optional<std::string> cache_dir = std::nullopt,
+        std::string load_format              = "auto",
+        bool fall_back_to_pt                 = true,
+        std::optional<std::string> revision  = "None") {
```

Comment:
Please make the default value of ``revision`` to ``std::nullopt``, as same as ``cache_dir``.